### PR TITLE
redo(cdc): custom redo event cache and disable it by default (#10139)

### PR DESCRIPTION
**Pull request by sdojjy**
close pingcap/tiflow#10143
# Conflicts:
#	cdc/processor/sinkmanager/manager.go

### What problem does this PR solve?

Issue Number: close #xxx

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: cdc/api/v2/model.go [hunk 1]
```diff
@@ -693,6 +706,14 @@ type ConsistentConfig struct {
 	UseFileBackend        bool   `json:"use_file_backend"`
 	Compression           string `json:"compression,omitempty"`
 	FlushConcurrency      int    `json:"flush_concurrency,omitempty"`
+
+	MemoryUsage *ConsistentMemoryUsage `json:"memory_usage"`
+}
+
+// ConsistentMemoryUsage represents memory usage of Consistent module.
+type ConsistentMemoryUsage struct {
+	MemoryQuotaPercentage uint64 `json:"memory_quota_percentage"`
```
Comment (nongfushanquan) [inline cdc/api/v2/model.go:715]: redo_memory_quota_percentage

**Review by nongfushanquan — APPROVED** [review]

File: dm/tests/lightning_mode/run.sh [hunk 2]
```diff
@@ -73,10 +73,10 @@ function run() {
 	run_tidb_server 4000 $TIDB_PASSWORD
 }
 
-cleanup_data lightning_mode
-# also cleanup dm processes in case of last run failed
-cleanup_process $*
-run $*
-cleanup_process $*
+#cleanup_data lightning_mode
```
Comment (CharlesCheung96) [inline dm/tests/lightning_mode/run.sh:76]: should such changes be reverted？

Comment (sdojjy) [reply dm/tests/lightning_mode/run.sh:76]: it seems these two dm integration tests need be fixed

File: cdc/api/v2/model.go (hunk 1 above)
Comment (hicqu) [reply cdc/api/v2/model.go:715]: The struct is named `ConsistentMemoryUsage`, with the `Consistent` prefix. so internal fields can be shorter.

**Review by hicqu — APPROVED** [review]

**Pull request merged**